Update Python AWS Lambda integration docs. (#8246)
getsentry/sentry-docs: src/platforms/python/common/integrations/aws-lambda/manual-instrumentation/index.mdx

---
title: AWS Lambda Manual Instrumentation
description: "Learn about instrumenting your AWS lambda function manually using Sentry."
sidebar_order: 2000
---

> <Note>
> 
> We also support
> 
> - [adding Sentry without modifying your code](/product/integrations/cloud-monitoring/aws-lambda/),
> - [installing Sentry in Lambda Layer](/platforms/python/integrations/aws-lambda/manual-layer/),
> - and [installing Sentry as a Container Image](/platforms/python/integrations/aws-lambda/container-image/).
> 
> </Note>

Reviewer: ```suggestion
<Note>

We also support:

- [Adding Sentry without modifying your code](/product/integrations/cloud-monitoring/aws-lambda/)
- [Installing Sentry in the Lambda Layer](/platforms/python/integrations/aws-lambda/manual-layer/)
- [Installing Sentry as a container image](/platforms/python/integrations/aws-lambda/container-image/)

</Note>
```

## Install

Create a deployment package on your local machine and install the required dependencies in the deployment package.
For more information, see [AWS Lambda deployment package in Python](https://docs.aws.amazon.com/lambda/latest/dg/python-package.html).

> Install our Python SDK using `pip` into your package directory:

Reviewer: ```suggestion
Install our Python SDK into your package directory using `pip`:
```

```bash
pip install --target ./package --upgrade sentry-sdk
```

## Configure

> You can use the AWS Lambda integration for the Python SDK like this:

Reviewer: ```suggestion
Here's how to use the AWS Lambda integration for the Python SDK:
```

<SignInNote />

```python
import sentry_sdk
from sentry_sdk.integrations.aws_lambda import AwsLambdaIntegration

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",
    integrations=[
        AwsLambdaIntegration(),
    ],

    # Set traces_sample_rate to 1.0 to capture 100%
    # of transactions for performance monitoring.
    # We recommend adjusting this value in production,
    traces_sample_rate=1.0,
)

def lambda_handler(event, context):
    # ...
```

Check out Sentry's [AWS sample apps](https://github.com/getsentry/examples/tree/master/aws-lambda/python) for detailed examples.

<Alert level="info" title="Note">

> 
> If you are using another web framework inside of AWS Lambda, the framework might catch those exceptions before we get to see them. Make sure to enable the framework specific integration as well, if one exists. See [_the Python main page_](/platforms/python/) for more information.
> 

Reviewer: ```suggestion

If you're using another web framework inside of AWS Lambda, that framework may catch exceptions before we see them. Make sure to enable the framework-specific integration if it exists. See [_the Python main page_](/platforms/python/) for more information.

```

</Alert>

## Verify

> Deriberately add an error to you function and run it. The error is now captured and sent to Sentry.io

Reviewer: ```suggestion
Add an error to you function and run it. If everything's working properly, it should be captured and sent to Sentry.io.
```

## Behavior

With the AWS Lambda integration enabled, the Python SDK will:

> - Automatically report all events from your Lambda Functions.
> - You can <PlatformLink to="/configuration/filtering/">modify the transaction sample rate</PlatformLink> using `traces_sample_rate`.

Reviewer: ```suggestion
- Automatically report all events from your Lambda Functions.
- Allow you to <PlatformLink to="/configuration/filtering/">modify the transaction sample rate</PlatformLink> using `traces_sample_rate`.
```

> - Event reports automatically include:
> 
>   - A link to CloudWatch Logs
>   - Function details
>   - `sys.argv` for the function
>   - AWS Request ID
>   - Function execution time
>   - Function version
>   - You can add more data as described <PlatformLink to="/enriching-events/">here</PlatformLink>

Reviewer: ```suggestion
- Automatically include the following event reports:

  - A link to CloudWatch Logs
  - Function details
  - `sys.argv` for the function
  - AWS Request ID
  - Function execution time
  - Function version
  
  You can add more data as described <PlatformLink to="/enriching-events/">here</PlatformLink>.
```

> - Request data is attached to all events: **HTTP method, URL, headers, form data, JSON payloads**. Sentry excludes raw bodies and multipart file uploads. Sentry also excludes personally identifiable information (such as user ids, usernames, cookies, authorization headers, IP addresses) unless you set `send_default_pii` to `True`.
> 

Reviewer: ```suggestion
- Request data including **HTTP method, URL, headers, form data, and JSON payloads** will be attached to all events. Sentry excludes raw bodies and multipart file uploads. Sentry also excludes personally identifiable information such as user ids, usernames, cookies, authorization headers, and IP addresses (unless you set `send_default_pii` to `True`).

```

- Each request has a separate scope. Changes to the scope within a view, for example setting a tag, will only apply to events sent as part of the request being handled.

## Options

> You can set options for `AwsLambdaIntegration` to change its behavior:

Reviewer: ```suggestion
You can change the behavior of `AwsLambdaIntegration` by setting options for it:
```
